metricsaggregator/src/main/java/io/dockstore/metricsaggregator/client/cli/TerraMetricsSubmitter.java

```diff
@@ -0,0 +1,572 @@
+    private final Map<String, List<MinimalWorkflowInfo>> workflowPathPrefixToWorkflows = new ConcurrentHashMap<>();
+
+    public TerraMetricsSubmitter(MetricsAggregatorConfig config, SubmitTerraMetrics submitTerraMetricsCommand) {
+        this.config = config;
+        this.submitTerraMetricsCommand = submitTerraMetricsCommand;
+    }
+
+    public void submitTerraMetrics() {
+        ApiClient apiClient = setupApiClient(config.getDockstoreServerUrl(), config.getDockstoreToken());
+        ExtendedGa4GhApi extendedGa4GhApi = new ExtendedGa4GhApi(apiClient);
+        WorkflowsApi workflowsApi = new WorkflowsApi(apiClient);
+
+        // Read CSV file
+        Iterable<CSVRecord> workflowMetricRecords;
+        final String inputDateFilePath = this.submitTerraMetricsCommand.getDataFilePath();
+        try (BufferedReader metricsBufferedReader = new BufferedReader(new FileReader(inputDateFilePath))) {
+            final CSVFormat csvFormat = CSVFormat.DEFAULT.builder()
+                    .setHeader(TerraMetricsCsvHeaders.class)
+                    .setSkipHeaderRecord(true)
+                    .setTrim(true)
+                    .build();
+            workflowMetricRecords = csvFormat.parse(metricsBufferedReader);
+
+            // This output file is used to record skipped executions
+            final String outputFileName = inputDateFilePath + "_skipped_executions_" + Instant.now().truncatedTo(ChronoUnit.SECONDS).toString().replace("-", "").replace(":", "") + ".csv";
+
+            List<CSVRecord> workflowMetricsToProcess = new ArrayList<>();
+            try (CSVPrinter skippedExecutionsCsvPrinter = submitTerraMetricsCommand.isRecordSkippedExecutions() ? new CSVPrinter(
+                    new FileWriter(outputFileName, StandardCharsets.UTF_8),
+                    CSVFormat.DEFAULT.builder().setHeader(SkippedTerraMetricsCsvHeaders.class).build()) : null) {
+
+                // Process the executions by reading 100,000 rows, then processing the rows in parallel
+                for (CSVRecord workflowMetricRecord: workflowMetricRecords) {
+                    final int batchSize = 100000;
+                    if (workflowMetricsToProcess.size() < batchSize && workflowMetricRecords.iterator().hasNext()) {
+                        workflowMetricsToProcess.add(workflowMetricRecord);
+                    } else {
+                        workflowMetricsToProcess.add(workflowMetricRecord);
+                        // Collect a map of CSV records with the same source URL
+                        Map<String, List<CSVRecord>> sourceUrlToCsvRecords = workflowMetricsToProcess.stream()
+                                .collect(groupingBy(csvRecord -> csvRecord.get(TerraMetricsCsvHeaders.source_url)));
+                        /*
+                        for (CSVRecord workflowMetricToProcess: workflowMetricsToProcess) {
```

Comment:
Could use comment if there is intent to reactivate (or could put in feature flag)

Reply:
Ah, missed this in my cleanup, this code isn't needed so I'll remove it